framework/web/AssetManager.php

```diff
@@ -572,10 +594,15 @@ public function getPublishedUrl($path)
      * Generate a CRC32 hash for the directory path. Collisions are higher
      * than MD5 but generates a much smaller hash string.
      * @param string $path string to be hashed.
+     * @param boolean $isFile whether the path is file.
      * @return string hashed string.
      */
-    protected function hash($path)
+    protected function hash($path, $isFile = true)
```

Comment:
`is_file` is a stat operation. I'd prefer something like `pathinfo($path, PATHINFO_EXTENSION) === ''` which won't touch FS.